bottlecap/src/metrics/enhanced/lambda.rs

```diff
@@ -233,29 +236,31 @@ mod tests {
 
         let mut ms_sketch = ddsketch_agent::DDSketch::default();
         ms_sketch.insert(0.1);
-        assert_eq!(
+        if let Some(ValueVariant::DDSketch(variant)) =
             aggr.get_value_by_id(constants::DURATION_METRIC.into(), None)
-                .unwrap(),
-            ms_sketch
-        );
-        assert_eq!(
+        {
+            assert_eq!(variant, ms_sketch);
+        }
+        if let Some(ValueVariant::DDSketch(variant)) =
             aggr.get_value_by_id(constants::BILLED_DURATION_METRIC.into(), None)
-                .unwrap(),
-            ms_sketch
-        );
+        {
+            assert_eq!(variant, ms_sketch);
+        }
+
         let mut mem_used_sketch = ddsketch_agent::DDSketch::default();
         mem_used_sketch.insert(128.0);
-        assert_eq!(
+        if let Some(ValueVariant::DDSketch(variant)) =
             aggr.get_value_by_id(constants::MAX_MEMORY_USED_METRIC.into(), None)
-                .unwrap(),
-            mem_used_sketch
-        );
+        {
+            assert_eq!(variant, mem_used_sketch);
+        }
+
         let mut max_mem_sketch = ddsketch_agent::DDSketch::default();
         max_mem_sketch.insert(256.0);
-        assert_eq!(
+        if let Some(ValueVariant::DDSketch(variant)) =
             aggr.get_value_by_id(constants::MEMORY_SIZE_METRIC.into(), None)
-                .unwrap(),
-            max_mem_sketch
-        );
+        {
+            assert_eq!(variant, max_mem_sketch);
+        }
```

Comment:
We should panic on else here, or else, if the value doesn't resolve into the expected, it will execute as a false positive.

Reply:
thanks, missed that!